qutebrowser/config/configtypes.py

```diff
@@ -1760,11 +1760,14 @@ def to_py(self, value: _StrUnset) -> _StrUnsetNone:
         elif not value:
             return None
 
-        if not ('{}' in value or '{0}' in value):
+        if not re.search('{(|0|quoted)}', value):
             raise configexc.ValidationError(value, "must contain \"{}\"")
 
         try:
-            value.format("")
+            format_keys = {
+                'quoted': "",
+            }
+            value.format("", **format_keys)
```

Comment:
There already is a `FormatString` type - I wonder, maybe `SearchEngineUrl` should have a `FormatString` member (or inherit from it, though I prefer composition) and use that? That already has a way to check which placeholders are allowed, so you'd only need to extend it to have a `requires_placeholder=True` (or so) flag (it probably could just search for `{.*}` or something then).

Reply:
For me (not knowing python and your code base well enough), this is too much work, because `FormatString` does not allow `{}`, so one also needs to take care of that.. (And I think this is something one can also do later)